cpp/core/utils/ObjectStore.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <set>
+#include "utils/ResourceMap.h"
+#include "utils/exception.h"
+
+namespace gluten {
+
+// A store for caching shared-ptrs and enlarging lifecycles of the ptrs to match lifecycle of the store itself by
+// default, and also serving release calls to release a ptr in advance. This is typically used in JNI scenario to bind
+// a shared-ptr's lifecycle to a Java-side object or some kind of resource manager.
+class ObjectStore {
+ public:
+  static std::unique_ptr<ObjectStore> create() {
+    return std::unique_ptr<ObjectStore>(new ObjectStore());
```

Comment:
why not use `std::make_unique` ?

Reply:
The ctor has private access. In the case `std::make_unique` doesn't work.